### PR TITLE
Improve error handing and error messages

### PR DESCRIPTION
**Pull request by mbudiu-vmw**

### CODE REVIEW

**Comment by fruffy** [comment]
Just saw this. I can remove the warning so the function can be removed. But what is a suitable replacement used by p4c?

**Comment by mbudiu-vmw** [comment]
Use `warn` with an additional ErrorType argument instead.

**Comment by fruffy** [comment]
Should be fixed now:
https://github.com/p4gauntlet/toz3/blob/master/codegen.cpp#L1081

**Comment by mbudiu-vmw** [comment]
Can you check that you got all of them by removing the function `warning` without this argument from the lib/error.h?

**Comment by fruffy** [comment]
Yeah everything looks good.

File: backends/bmv2/psa_switch/psaSwitch.h [hunk 1]
```diff
@@ -187,6 +187,11 @@ class ParsePsaArchitecture : public Inspector {
     explicit ParsePsaArchitecture(PsaProgramStructure* structure) :
         structure(structure) { CHECK_NULL(structure); }
 
+    void modelError(const char* format, const IR::INode* node) {
+        ::error(format, node->getNode());
```
Comment (ChrisDodd) [inline backends/bmv2/psa_switch/psaSwitch.h:191]: should perhaps be `ErrorType::ERR_INVALID`?  Or perhaps a new ERR_ARCH_INTERNAL or some such

File: control-plane/bytestrings.cpp [hunk 2]
```diff
@@ -28,7 +28,7 @@ namespace ControlPlaneAPI {
 boost::optional<std::string> stringReprConstant(big_int value, int width) {
     // TODO(antonin): support negative values
     if (value < 0) {
-        ::error("%1%: Negative values not supported yet", value);
+        ::error(ErrorType::ERR_INVALID, "%1%: Negative values not supported yet", value);
```
Comment (ChrisDodd) [inline control-plane/bytestrings.cpp:31]: probably should be ERR_UNSUPPORTED

**Review by ChrisDodd — APPROVED** [review]

**Comment by mbudiu-vmw** [comment]
@osinstom : there are a bunch of calls to `::error()` in your code that don't have an error code; when you have some time please convert these as well.

**Pull request merged**